### ISSUE TITLE
BUG: Issue with Ordered Transform in Ordered Logistic API docs example

### ISSUE THREAD

**Issue by NathanielF** [pymc-devs/pymc#6610]
### Describe the issue:

The API docs for the ordered logistic class recommends using the transform ordered to provide cutpoints for the ordinal regression. But the provided example breaks with an error reporting that the random variable for the cutpoints lacks a shape. 

<img width="880" alt="image" src="https://user-images.githubusercontent.com/10588167/226060429-f091171e-b00c-4354-9a01-afca3d59fdad.png">

On the latest version: 
<img width="410" alt="image" src="https://user-images.githubusercontent.com/10588167/226060508-5fa6260a-c1d8-4090-b436-a5fe23356fb2.png">

### Reproduceable code example:

```python
import arviz as az
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import pymc as pm
import numpy as np
import pytensor as pt

# Generate data for a simple 1 dimensional example problem
n1_c = 300; n2_c = 300; n3_c = 300
cluster1 = np.random.randn(n1_c) + -1
cluster2 = np.random.randn(n2_c) + 0
cluster3 = np.random.randn(n3_c) + 2

x = np.concatenate((cluster1, cluster2, cluster3))
y = np.concatenate((1*np.ones(n1_c),
                    2*np.ones(n2_c),
                    3*np.ones(n3_c))) - 1

# Ordered logistic regression
with pm.Model() as model:
    cutpoints = pm.Normal("cutpoints", mu=[-1,1], sigma=10, shape=2, 
                          transform=pm.distributions.transforms.Ordered)
    y_ = pm.OrderedLogistic("y", cutpoints=cutpoints, eta=x, observed=y)
    idata = pm.sample()
```

### Error message:

```shell
Output exceeds the size limit. Open the full output data in a text editor
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[10], line 14
     12 # Ordered logistic regression
     13 with pm.Model() as model:
---> 14     cutpoints = pm.Normal("cutpoints", mu=[-1,1], sigma=10, shape=2, 
     15                           transform=pm.distributions.transforms.Ordered)
     16     y_ = pm.OrderedLogistic("y", cutpoints=cutpoints, eta=x, observed=y)
     17     idata = pm.sample()

File ~/mambaforge/envs/pymc_examples_new/lib/python3.11/site-packages/pymc/distributions/distribution.py:312, in Distribution.__new__(cls, name, rng, dims, initval, observed, total_size, transform, *args, **kwargs)
    308         kwargs["shape"] = tuple(observed.shape)
    310 rv_out = cls.dist(*args, **kwargs)
--> 312 rv_out = model.register_rv(
    313     rv_out,
    314     name,
    315     observed,
    316     total_size,
    317     dims=dims,
    318     transform=transform,
    319     initval=initval,
    320 )
    322 # add in pretty-printing support
    323 rv_out.str_repr = types.MethodType(str_for_dist, rv_out)
...
---> 95     y = at.zeros(value.shape)
     96     y = at.inc_subtensor(y[..., 0], value[..., 0])
     97     y = at.inc_subtensor(y[..., 1:], at.log(value[..., 1:] - value[..., :-1]))

AttributeError: 'RandomGeneratorSharedVariable' object has no attribute 'shape'
```

### PyMC version information:

Last updated: Fri Mar 17 2023

Python implementation: CPython
Python version       : 3.11.0
IPython version      : 8.11.0

pytensor: 2.10.1

numpy     : 1.24.2
arviz     : 0.15.1
matplotlib: 3.7.1
pandas    : 1.5.3
pymc      : 5.1.1
pytensor  : 2.10.1

Watermark: 2.3.1

### Context for the issue:

I was going to try and write up docs on the technique of ordinal regression, but i think failure of the ordered transform makes the entire class of models less straightforward to implement. I'm pretty sure it's related to this line: https://github.com/pymc-devs/pymc/blob/c7279b5e3f83823c068c2689075f016a361bb472/pymc/distributions/transforms.py#L89

But i don't know enough about the random variable implementation to know why the shape attribute is not available now.

**Comment by ricardoV94** [comment]
Can you share the whole traceback?

**Comment by NathanielF** [comment]
`---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[47], line 14
     12 # Ordered logistic regression
     13 with pm.Model() as model:
---> 14     cutpoints = pm.Normal("cutpoints", mu=[-1,1], sigma=10, shape=2,
     15                           transform=pm.distributions.transforms.Ordered)
     16     y_ = pm.OrderedLogistic("y", cutpoints=cutpoints, eta=x, observed=y)
     17     idata = pm.sample()

File ~/mambaforge/envs/pymc_examples_new/lib/python3.11/site-packages/pymc/distributions/distribution.py:312, in Distribution.__new__(cls, name, rng, dims, initval, observed, total_size, transform, *args, **kwargs)
    308         kwargs["shape"] = tuple(observed.shape)
    310 rv_out = cls.dist(*args, **kwargs)
--> 312 rv_out = model.register_rv(
    313     rv_out,
    314     name,
    315     observed,
    316     total_size,
    317     dims=dims,
    318     transform=transform,
    319     initval=initval,
    320 )
    322 # add in pretty-printing support
    323 rv_out.str_repr = types.MethodType(str_for_dist, rv_out)

File ~/mambaforge/envs/pymc_examples_new/lib/python3.11/site-packages/pymc/model.py:1328, in Model.register_rv(self, rv_var, name, observed, total_size, dims, transform, initval)
   1326     raise ValueError("total_size can only be passed to observed RVs")
   1327 self.free_RVs.append(rv_var)
-> 1328 self.create_value_var(rv_var, transform)
   1329 self.add_named_variable(rv_var, dims)
   1330 self.set_initval(rv_var, initval)

File ~/mambaforge/envs/pymc_examples_new/lib/python3.11/site-packages/pymc/model.py:1521, in Model.create_value_var(self, rv_var, transform, value_var)
   1518         value_var.tag.test_value = rv_var.tag.test_value
   1519 else:
   1520     # Create value variable with the same type as the transformed RV
-> 1521     value_var = transform.forward(rv_var, *rv_var.owner.inputs).type()
   1522     value_var.name = f"{rv_var.name}_{transform.name}__"
   1523     value_var.tag.transform = transform

File ~/mambaforge/envs/pymc_examples_new/lib/python3.11/site-packages/pymc/distributions/transforms.py:95, in Ordered.forward(self, value, *inputs)
     94 def forward(self, value, *inputs):
---> 95     y = at.zeros(value.shape)
     96     y = at.inc_subtensor(y[..., 0], value[..., 0])
     97     y = at.inc_subtensor(y[..., 1:], at.log(value[..., 1:] - value[..., :-1]))

AttributeError: 'RandomGeneratorSharedVariable' object has no attribute 'shape'`

**Comment by ricardoV94** [comment]
Ah maybe Ordered is not an initialized transform? Try to use `distributions.transforms.univariate_ordered` or `Ordered()` instead

**Comment by ricardoV94** [comment]
Ah maybe Ordered is not an initialized transform? Try to use `distributions.transforms.univariate_ordered` or `...ordered` instead.

I think the docstring example uses the latter (small `o` for the instance instead of big `O` for the class)

**Comment by NathanielF** [comment]
That looks right now: 
<img width="944" alt="image" src="https://user-images.githubusercontent.com/10588167/226092603-82c77301-f9ee-40d0-a479-76e3a3413fd7.png">

<img width="990" alt="image" src="https://user-images.githubusercontent.com/10588167/226092626-d10aa3a9-e6b0-437b-8d64-4da53eab1d6e.png">

**Comment by ricardoV94** [comment]
The warning might be a bug we fix just now in pymc5.1.2

Not sure what the plot tells you? Good or bad?

**Comment by ricardoV94** [comment]
Ah you said it looks right. Shall we close the issue?

Maybe we can update the docs to mention `univariate_ordered`?

This way reduces the chance of typos, and it's actually the right one going forward.

**Comment by NathanielF** [comment]
I think the plot looks good. Yes, i'll open a PR now to change the docstring

**Comment by ricardoV94** [comment]
Closed via #6611
